cdc/kv/matcher.go

```diff
@@ -39,39 +36,60 @@ func newMatchKey(row *cdcpb.Event_Row) matchKey {
 
 func newMatcher() *matcher {
 	return &matcher{
-		unmatchedValue: make(map[matchKey]*pendingValue),
+		unmatchedValue: make(map[matchKey]*cdcpb.Event_Row),
 	}
 }
 
 func (m *matcher) putPrewriteRow(row *cdcpb.Event_Row) {
 	key := newMatchKey(row)
-	value := row.GetValue()
-	oldvalue := row.GetOldValue()
-	// tikv may send a prewrite event with empty value (txn heartbeat)
-	// here we need to avoid the invalid prewrite event overwrite the value
-	if _, exist := m.unmatchedValue[key]; exist && len(value) == 0 {
+	// tikv may send a fake prewrite event with empty value caused by txn heartbeat.
+	// here we need to avoid the fake prewrite event overwrite the prewrite value.
+
+	// when the old-value is disabled, the value of the fake prewrite event is empty.
+	// when the old-value is enabled, the value of the fake prewrite event is also empty,
+	// but the old value of the fake prewrite event is not empty.
+	// We can distinguish fake prewrite events by whether the value is empty,
+	// no matter the old-value is enable or disabled
+	if _, exist := m.unmatchedValue[key]; exist && len(row.GetValue()) == 0 {
 		return
 	}
-	m.unmatchedValue[key] = &pendingValue{
-		value:    value,
-		oldValue: oldvalue,
-	}
+	m.unmatchedValue[key] = row
 }
 
-func (m *matcher) matchRow(row *cdcpb.Event_Row) (*pendingValue, bool) {
+func (m *matcher) matchRow(row *cdcpb.Event_Row) bool {
```

Comment:
Please add a comment that the `Value` and `OldValue` will be assigned if a matched prewrite event exists.